### PR TITLE
Update tax form queries for cross-host expenses

### PR DESCRIPTION
**Pull request by Betree**
Resolve https://github.com/opencollective/opencollective/issues/5068

### CODE REVIEW

File: server/lib/queries.js [hunk 1]
```diff
@@ -1017,9 +1021,13 @@ const getTaxFormsRequiredForExpenses = async expenseIds => {
       ON all_expenses_collectives.id = all_expenses."CollectiveId"
       AND all_expenses_collectives."HostCollectiveId" = d."HostCollectiveId"
     LEFT JOIN "LegalDocuments" ld
-      ON ld."CollectiveId" = analyzed_expenses."FromCollectiveId"
-      AND ld.year + :validityInYears >= date_part('year', analyzed_expenses."incurredAt")
+      ON ld.year + :validityInYears >= date_part('year', analyzed_expenses."incurredAt")
       AND ld."documentType" = 'US_TAX_FORM'
+      AND ld."requestStatus" = 'RECEIVED'
```
Comment (Betree) [inline server/lib/queries.js:1026]: Here I'm filtering to get only the received documents to ignore all expenses when a doc is available - either on the account or the host - with the `AND ld.id IS NULL` below.

File: test/server/lib/tax-forms.test.js [hunk 2]
```diff
@@ -140,6 +143,13 @@ describe('server/lib/tax-forms', () => {
       year: year - 1,
     });
 
+    // Create legal document for accountWithTaxFormSubmittedByHost (no tax form should be required in this case)
+    await fakeLegalDocument({
+      CollectiveId: accountWithTaxFormSubmittedByHost.HostCollectiveId,
+      requestStatus: 'RECEIVED',
+      year: year,
+    });
+
```
Comment (Betree) [inline test/server/lib/tax-forms.test.js:152]: The goal of this chunk + the `Expense.create` below is to make sure `accountWithTaxFormSubmittedByHost` is not included in the test `findAccountsThatNeedToBeSentTaxForm` > `returns the right profiles`

File: server/lib/queries.js [hunk 3]
```diff
@@ -1029,9 +1037,10 @@ const getTaxFormsRequiredForExpenses = async expenseIds => {
     AND analyzed_expenses."deletedAt" IS NULL
     AND (from_collective."HostCollectiveId" IS NULL OR from_collective."HostCollectiveId" != c."HostCollectiveId")
     AND all_expenses.type NOT IN (:ignoredExpenseTypes)
-    AND all_expenses.status NOT IN ('ERROR', 'REJECTED', 'DRAFT', 'UNVERIFIED')
+    AND all_expenses.status NOT IN (:ignoredExpenseStatuses)
```
Comment (znarf) [inline server/lib/queries.js:1040]: Looking good.

**Review by znarf — APPROVED** [review]
Thank you! Looking good.

**Pull request merged**